pkg/pillar/pubsub/util.go

```diff
@@ -106,3 +106,34 @@ func ConnReadCheck(conn net.Conn) error {
 	}
 	return sysErr
 }
+
+// ChannelWatch describe a channel to watch and the callback to call
+type ChannelWatch struct {
+	// Chan is the channel to watch for incoming data
+	Chan reflect.Value
```

Comment:
Would it be easier to use some chan interface type for this?
Still need to type cast but ...

Reply:
as I understand it is not possible to cast a channel: https://stackoverflow.com/a/25594926